app/models/concerns/searchable.rb

```diff
@@ -71,17 +71,22 @@ def index_names(sources = nil)
       models.map(&:index_name)
     end
 
-    def fetch_all
+    def fetch_all(format = :csv)
       search_options = { scroll: '5m', search_type: 'scan', index: index_names }
       search_options[:type] = index_type if index_type
       response = ES.client.search(search_options)
 
-      results = []
-
+      results = format == :json ? { offset: 0 } : []
       while response = ES.client.scroll(scroll_id: response['_scroll_id'], scroll: '5m')
         break if response['hits']['hits'].empty?
-        batch = response['hits']['hits'].map(&:deep_symbolize_keys)
-        results.push(*batch)
+        case format
+          when :json
+            batch = response['hits'].deep_symbolize_keys
+            results.merge!(batch)
+          else
+            batch = response['hits']['hits'].map(&:deep_symbolize_keys)
+            results.push(*batch)
+        end
```

Comment:
Here's how I see this code: `fetch_all` is responsible for fetching all documents from an index. It doesn't care about the requested output format, only that the caller wants all documents. Passing the requested format is polluting the simple nature of the method with information it shouldn't need.

But you are passing it for a reason. The reason is that you need the return value's format to match the [`search_for`](https://github.com/GovWizely/webservices/pull/174/files#diff-7ae1034585424914bae40382262f2e59R21). You want to take the return value from either method and pass it to the view.

A cleaner way to do this is to adjust the return value of `fetch_all` to match `search_for`, then adjust the CSV code to use this new format. Please do that instead.